### ISSUE TITLE
segfault in mk_unit_resolution

### ISSUE THREAD

**Issue by cheshire** [Z3Prover/z3#51]
Hi,

I encounter a segfault inside mk_unit_resolution.

Unfortunately, the replayable log I have for the problem is rather large (143MB): http://metaworld.me/segfault.log .
I can compile Z3 in debug mode and look at the stack trace to try to isolate the problem if required.

**Comment by NikolajBjorner** [comment]
Is this for the opt branch? In this case it seems you are enabling proof mode, which should not be necessary.

**Comment by cheshire** [comment]
Indeed. Sure, I'll disable it, I just recall having segfaults with PROOF mode disabled and no performance benefits, but perhaps the situation has changed.
Thanks!

**Comment by NikolajBjorner** [comment]
Hopefully the segfaults you reported earlier without proof mode have been addressed. At least I looked at them and found problems with the difference logic engines for optimization. The behavior with proof mode is that Z3 uses the Simplex solver. The default solver selection has also changed somewhat recently, so it should pick the simplex engine unless instructed otherwise. The crash in mk_unit_resolution should of course be addressed, but hopefully this is not necessary for this use.
